src/components/SingleEvent/EventFilters.js

```diff
@@ -22,28 +21,52 @@ const Filter = styled('div')`
 
 const EventFiltersContainer = styled('div')``
 
+const QRScannerContainer = styled('div')`
+  margin-top: 20px;
+  width: 100%;
+`
+
+const CenteredQrReader = styled(QrReader)`
+  margin-top: 20px;
+  margin-left: auto;
+  margin-right: auto;
+  width: 75%;
+`
+
 class EventFilters extends Component {
-  state = {}
-
-  _scan = client => () => {
-    this.setState({ scanError: null }, async () => {
-      try {
-        const { error, data = {} } = await client.query({
-          query: QR_QUERY,
-          fetchPolicy: 'no-cache'
-        })
-
-        if (error) {
-          throw error
-        }
+  constructor(props) {
+    super(props)
 
-        if (data.qrCode) {
-          this.props.handleSearch(data.qrCode)
+    this.state = {}
+
+    if (navigator && navigator.permissions) {
+      navigator.permissions.query({ name: 'camera' }).then(result => {
+        if (result.state === 'denied') {
+          this._disableCamera()
         }
-      } catch (scanError) {
-        this.setState({ scanError })
-      }
-    })
+      })
+    }
+  }
+
+  _disableCamera = () => {
+    this.setState({ cameraUnavailable: true })
+  }
+
+  _scan = data => {
+    console.log(data)
+    if (!data) {
+      console.log('No data')
```

Comment:
Replace logs with a message to return to the user.

Remove the other logs unless you think they are useful for something other than debugging your own code